src/pages/node-express/lesson11-a1.mdx

```diff
@@ -0,0 +1,948 @@
+  const register = document.getElementById("register");
+
+  loginRegisterDiv.addEventListener("click", (e) => {
+    if (inputEnabled && e.target.nodeName === "BUTTON") {
+      if (e.target === login) {
+        showLogin();
+      } else if (e.target === register) {
+        showRegister();
+      }
+    }
+  });
+};
+
+export const showLoginRegister = () => {
+  setDiv(loginRegisterDiv);
+};
+```
+
+Each of the div handling modules follow this pattern. Required imports (used when one div handler
+calls another) are resolved up front. Then, within the handler function, the div and its controls are defined.
+Also, within the handler function, an event handler is declared to handle mouse clicks
+within the div.
+
+A separate function handles display of the div. (React works in similar fashion, if you know that framework -- but this lesson does not use React.)
+
+The `register.js` module is as follows:
+
+```js
+import {
+  inputEnabled,
+  setDiv,
+  message,
+  token,
+  enableInput,
+  setToken,
+} from "./index.js";
+import { showLoginRegister } from "./loginRegister.js";
+import { showJobs } from "./jobs.js";
+
+let registerDiv = null;
+let name = null;
+let email1 = null;
+let password1 = null;
+let password2 = null;
+
+export const handleRegister = () => {
+  registerDiv = document.getElementById("register-div");
+  name = document.getElementById("name");
+  email1 = document.getElementById("email1");
+  password1 = document.getElementById("password1");
+  password2 = document.getElementById("password2");
+  const registerButton = document.getElementById("register-button");
+  const registerCancel = document.getElementById("register-cancel");
+
+  registerDiv.addEventListener("click", (e) => {
+    if (inputEnabled && e.target.nodeName === "BUTTON") {
+      if (e.target === registerButton) {
+        showJobs();
+      } else if (e.target === registerCancel) {
+        showLoginRegister();
+      }
+    }
+  });
+};
+
+export const showRegister = () => {
+  email1.value = null;
+  password1.value = null;
+  password2.value = null;
+  setDiv(registerDiv);
+};
+```
+
+The `login.js` module is as follows:
+
+```js
+import {
+  inputEnabled,
+  setDiv,
+  token,
+  message,
+  enableInput,
+  setToken,
+} from "./index.js";
+import { showLoginRegister } from "./loginRegister.js";
+import { showJobs } from "./jobs.js";
+
+let loginDiv = null;
+let email = null;
+let password = null;
+
+export const handleLogin = () => {
+  loginDiv = document.getElementById("logon-div");
+  email = document.getElementById("email");
+  password = document.getElementById("password");
+  const logonButton = document.getElementById("logon-button");
+  const logonCancel = document.getElementById("logon-cancel");
+
+  loginDiv.addEventListener("click", (e) => {
+    if (inputEnabled && e.target.nodeName === "BUTTON") {
+      if (e.target === logonButton) {
+        showJobs();
+      } else if (e.target === logonCancel) {
+        showLoginRegister();
```

Comment:
In the existing page, we show the clearing of the inputs here; but in my update we only introduce that a bit later down at the same time as we do it for the register page.